src/CodeGen_Hexagon.cpp

```diff
@@ -455,6 +455,9 @@ CodeGen_Hexagon::CodeGen_Hexagon(Target t)
 
 std::unique_ptr<llvm::Module> CodeGen_Hexagon::compile(const Module &module) {
     auto llvm_module = CodeGen_Posix::compile(module);
+    static bool options_processed = false;
+    if (options_processed) return llvm_module;
```

Comment:
Note that this skips both the hexagon-small-data-threshold and the enable-hexagon-hvx-double options, so the hexagon-hvx-double option will only get set on the first Module that gets compiled.